keras/layers/containers.py

```diff
@@ -614,3 +614,34 @@ def set_weights(self, weights):
             nb_param = len(layer.get_weights())
             layer.set_weights(weights[:nb_param])
             weights = weights[nb_param:]
+
+    def get_named_weights(self):
```

Comment:
Is there any way to make it possible for `set_weights` to automatically detect the weight format, and for `get_weights` to take a `named` argument? It seems somewhat redundant to have the same set of methods twice.